controllers/dbaas.redhat.com/crunchybridgeinstance_controller.go

```diff
@@ -0,0 +1,311 @@
+					logger.Error(statusErr, "Error in updating CrunchyBridgeInstance status")
+					return ctrl.Result{Requeue: true}, statusErr
+				}
+
+				return ctrl.Result{}, err
+			}
+
+			// Assuming the request was sent, update phase
+			instanceObj.Status.Phase = crunchybridgev1alpha1.PhaseCreating
+			//instanceObj.Status.Updated = time.Now().Format(time.RFC3339)
+			if err := r.Status().Update(ctx, instanceObj); err != nil {
+				return ctrl.Result{}, err
+			}
+
+		case crunchybridgev1alpha1.PhaseCreating:
+			var detC bridgeapi.ClusterDetail
+			if cid := instanceObj.Status.InstanceID; cid == "" {
+				c, err := bridgeapiClient.ClusterByName(instanceObj.Spec.Name)
+				if err != nil {
+					return ctrl.Result{}, err
+				}
+				detC = c
+			} else {
+				c, err := bridgeapiClient.ClusterDetail(cid)
+				if err != nil {
+					return ctrl.Result{}, err
+				}
+				detC = c
+			}
+			logger.Info("cluster creating", "name", instanceObj.Spec.Name)
+
+			if err := r.updateStatusFromDetail(detC, &instanceObj.Status); err != nil {
+				statusErr := r.updateStatus(instanceObj, metav1.ConditionFalse, BackendError, err.Error())
+				if statusErr != nil {
+					logger.Error(statusErr, "Error in updating CrunchyBridgeInstance status")
+					return ctrl.Result{Requeue: true}, statusErr
+				}
+				return ctrl.Result{}, err
+			}
+
+			if readyNow := (detC.State == string(bridgeapi.StateReady)); readyNow {
+				instanceObj.Status.Phase = crunchybridgev1alpha1.PhaseReady
+				statusErr := r.updateStatus(instanceObj, metav1.ConditionTrue, Ready, InstanceSuccessMessage)
+				if statusErr != nil {
+					logger.Error(statusErr, "Error in updating CrunchyBridgeInstance status")
+					return ctrl.Result{Requeue: true}, statusErr
+				}
+				logger.Info("cluster created", "name", instanceObj.Spec.Name)
+			}
+
+			//instanceObj.Status.Updated = time.Now().Format(time.RFC3339)
+			if err := r.Status().Update(ctx, instanceObj); err != nil {
+				return ctrl.Result{}, err
+			}
+			return ctrl.Result{Requeue: true, RequeueAfter: WatchInt}, nil
+
+		case crunchybridgev1alpha1.PhaseReady:
+			// TODO: Monitor changes, change state machine (phoenix)
+
+		default:
+			return ctrl.Result{}, fmt.Errorf("unrecognized phase: %s", instanceObj.Status.Phase)
+		}
+
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *CrunchyBridgeInstanceReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&dbaasredhatcomv1alpha1.CrunchyBridgeInstance{}).
+		Complete(r)
+}
+
+func listContains(list []string, s string) bool {
+	for _, str := range list {
+		if str == s {
+			return true
+		}
+	}
+	return false
+}
+
+func (r *CrunchyBridgeInstanceReconciler) createFromSpec(spec dbaasv1alpha1.DBaaSInstanceSpec, bridgeapiClient *bridgeapi.Client) (bridgeapi.CreateRequest, error) {
+	var req bridgeapi.CreateRequest
+
+	req.Name = spec.Name
+	req.Provider = spec.CloudProvider
```

Comment:
If the provider or region is not provided by the user in the CR, should an error be raised here?

Reply:
for crunchy suppose to provide these values, as no trial version option, we need to enable the validation in the first place-> CRD validation or webhook.